FactorioWebInterface/Data/DefaultAdminAccount.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Serilog;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace FactorioWebInterface.Data
+{
+    public class DefaultAdminAccount
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly string _userName = "Admin";
+
+        public DefaultAdminAccount(UserManager<ApplicationUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        private enum AccountsNumbers
+        {
+            NoAccounts,
+            OnlyDefaultAccount,
+            DefaultIsOnlyRootAccount,
+            MultipleAccounts
+        }
+
+        public async Task SetupDefaultUserAsync()
+        {
+            string id = Constants.DefaultAdminAccount;
+
+            await ClearDefaultUserAsync(id);
+
+            switch (await OnlyAccount())
+            {
+                case AccountsNumbers.DefaultIsOnlyRootAccount:
+                case AccountsNumbers.OnlyDefaultAccount:
+                    Log.Information(Constants.DefaultAdminAccount + " could not be created, another account already exists");
+                    return;
+                case AccountsNumbers.MultipleAccounts:
+                    await ClearDefaultUserAsync(id, true);
+                    return;
+                case AccountsNumbers.NoAccounts:
+                    break;
+            }
+
+            await CreateDefaultUserAsync(id);
+        }
+
+        //Todo: Perform count directly on database
+        private async Task<AccountsNumbers> OnlyAccount()
+        {
+
+            var rootUsers = await _userManager.GetUsersInRoleAsync(Constants.RootRole);
+            int rootCount = rootUsers.Count();
+
+            if (rootCount == 1 && await ValidateDefaultUserAsync(rootUsers.First()))
+            {
+                return AccountsNumbers.DefaultIsOnlyRootAccount;
+            }
+
+            var users = _userManager.Users;
+            int userCount = users.Count();
+            if (userCount == 1 && await ValidateDefaultUserAsync(users.First())) 
+            {
+                return AccountsNumbers.OnlyDefaultAccount;
+            }
+            
+            if (userCount > 0 && rootCount > 0)
+            {
+                return AccountsNumbers.MultipleAccounts;
+            }
+
+            return AccountsNumbers.NoAccounts;
+        }
+
+        private async Task ClearDefaultUserAsync(string id, bool force = false)
+        {
+            ApplicationUser userResult = await _userManager.FindByIdAsync(id);
+            if (await ValidateDefaultUserAsync(userResult) && !force)
+            {
+                Log.Information("Valid " + Constants.DefaultAdminAccount + " already exists");
```

Comment:
For the logging I prefer using template strings <https://github.com/serilog/serilog/wiki/Formatting-Output>, or see other places where I've using logging.
This way the string concatenation is done by the logging thread. Which is obviously not needed in this case, but I value the consistency of doing it everywhere.